### DISCUSSION TITLE
"Database doesn't have Directus tables installed" but is there

### DISCUSSION THREAD

**Discussion by rokdd** [directus/directus#14304] [Archive]
From running directus locally I am trying to migrate it to a container. Together with this step I am also changing from sqlite to postgresql.

I am able to start the docker container and first it did not created a database. So what I did, that I connected locally to the remote postgres-db und init the database (would be also interesting to know what is correct way in a dockerimage to do this). Now when I start I get the following message:
`Database doesn't have Directus tables installed.`

So my database and the testuser exists, and when I login in db directly with same credentials I can update and edit normally. When I execute commands for `database init` or `bootstrap` it always says that either it already exists or the Directus tables are not installed.

What I am doing wrong?

**Comment by br41nslug** [comment]
Are you using the provided docker image or rolling your own? Sounds like there could be a problem with the environment variables for the running container.

**Reply by rijkvanzanten** [reply]
Sounds to me like the docker image is trying to connect to a different database than the one that has the tables installed 🤔

**Reply by rokdd** [reply]
@br41nslug I am running my own.
What is wired that the connection is working because else the error would be different.
@rijkvanzanten So I cannot use the same db which I created from local at the remote postgres for the docker?

**Reply by rijkvanzanten** [reply]
> What is wired that the connection is working because else the error would be different.

_A_ connection is working, but that doesn't mean it's connected to the correct database necessarily 🙂 

> @rijkvanzanten So I cannot use the same db which I created from local at the remote postgres for the docker?

You should be able to. Just saying that in this case it sounds like the Docker container isn't connected to the correct database somehow

**Reply by rokdd** [reply]
Okay I tried to init the project locally with the remote database and then directly afterwards started locally directus, it prints me the same error:
```

PS C:\Coding\R repos\xxx> npm init directus-project example-project
? Database Host: postgres.xxx.rds.amazonaws.com
? Port: 5432
? Database Name: test_db
? Database User: test_user
? Database Password: ******************************       
? Enable SSL: No

Create your first admin user:
Your project has been created at C:\Coding\R repos\xxx\example-project.
The configuration can be found in C:\Coding\R repos\xxx\example-project/.env

Start Directus by running:
  cd C:\Coding\R repos\xxx\example-project
  npx directus start
PS C:\Coding\R repos\xxx\example-project> npx directus start
12:59:35 ÔÜá´©Å  PUBLIC_URL should be a full URL
12:59:36 ÔÜá´©Å  PostGIS isn't installed. Geometry type support will be limited.
12:59:36 ­ƒÜ¿ Database doesn't have Directus tables installed.

```

**Comment by rokdd** [comment]
So finally I enabled trace logging (debug was not giving any more information as info) and saw that pg_class like the postgres catalogs was the last request to db. I ended up that appaerently the automatically generated databases and public schema has not enough permission (and we are not allowed to change). So I kind of used this options to run it:

```
DB_DATABASE="pg_xxx_directus_stable"
DB_SCHEMA="pg_xxx_directus_stable"
DB_CUSTOM_SCHEMA="pg_xxx_directus_stable"
DB_SEARCH_PATH="array:pg_xxx_directus_stable,public"
```
Thanks all for your help and the wonderful community!
